pyproject.toml

```diff
@@ -58,7 +57,7 @@ keywords = [
 license = "Apache-2.0"
 name = "dpctl"
 readme = {file = "README.md", content-type = "text/markdown"}
-requires-python = ">=3.9"
+requires-python = ">=3.10"
```

Comment:
Do we need to update the example in `examples/cython/sycl_buffer/README.md`?
It has been run with python 3.9.